liblustreapi-sys/build.rs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2019 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+macro_rules! mtime {
+    ($path:expr) => {
+        std::fs::metadata($path).unwrap().modified().unwrap()
+    };
+}
+
+fn main() {
+    let out_file = std::env::current_dir()
+        .unwrap()
+        .join("src")
+        .join("bindings.rs");
+
+    // Tell cargo to tell rustc to link the system liblustreapi
+    // shared library.
+    println!("cargo:rustc-link-lib=lustreapi");
+
+    if out_file.exists() && mtime!(&out_file) > mtime!(std::env::current_exe().unwrap()) {
```

Comment:
Checking for mtime as a guard means things like `cargo check` can't run if the header files don't exist.

It may not be obvious, but out_file by itself is a guard that allows compilation for other envs. Here's an example trying to compile this change on mac with mtime:

<img width="1760" alt="Screen Shot 2019-04-01 at 9 36 33 AM" src="https://user-images.githubusercontent.com/458717/55331745-ed38e300-5461-11e9-8ca6-20787dd68e76.png">

Reply:
I'm trying to get it to rebuild if build.rs updated (similar to logic of make)

Reply:
Understood, my larger point is we need to be able to build this workspace on other OSes than linux.